numba_dpex/core/descriptor.py

```diff
@@ -40,13 +41,19 @@ class DpexTargetOptions(CPUTargetOptions):
     release_gil = _option_mapping("release_gil")
     no_compile = _option_mapping("no_compile")
     use_mlir = _option_mapping("use_mlir")
+    inline_threshold = _option_mapping("inline_threshold")
+    _compilation_mode = _option_mapping("_compilation_mode")
 
     def finalize(self, flags, options):
         super().finalize(flags, options)
         _inherit_if_not_set(flags, options, "experimental", False)
         _inherit_if_not_set(flags, options, "release_gil", False)
         _inherit_if_not_set(flags, options, "no_compile", True)
         _inherit_if_not_set(flags, options, "use_mlir", False)
+        _inherit_if_not_set(flags, options, "inline_threshold", 0)
```

Comment:
```suggestion
        _inherit_if_not_set(flags, options, "inline_threshold", config.INLINE_THRESHOLD)
```

Let's set it here based on the config value to avoid code duplicate and complex logic in `_optimize_final_module`. Should we warn if both options set, so user knows that env variable does not take effect on particular function?